src/shared/routes.ts

```diff
@@ -67,12 +67,12 @@ export const routes: IRoutePropsWithFetch<Record<string, any>>[] = [
     fetchInitialData: Communities.fetchInitialData,
   },
   {
-    path: `/post/:post_id`,
+    path: `/post/:post_id/comment/:comment_id`,
     component: Post,
     fetchInitialData: Post.fetchInitialData,
   },
   {
-    path: `/comment/:comment_id`,
```

Comment:
Removing this would break all previous comment links. Plus the GetComment endpoint should correctly get the post and context.

I'll take a crack at fixing this, I'm not sure where this bug got introduced, because it worked well before `0.18.3`

Reply:
[Someone discussing the issue pointed out the problem.](https://github.com/LemmyNet/lemmy-ui/issues/1999#issuecomment-1664724892) It turns out the issue was because the post ID was not being passed to `getComments`. Hopefully there is a way to fix this without breaking links, but the URL change was motivated by the need to have both the post ID and the comment ID.

I may be able to accomplish this without changing the path if I utilize history state instead.

Reply:
Sure, go right ahead.

Reply:
Ugh, I can't even replicate the issue locally. Its probably because the root issue is this: https://github.com/LemmyNet/lemmy/pull/3717

Comments are now coming back in a random order, and our tree building code everywhere isn't resilient enough to handle unordered comments.

Reply:
Okay I have a PR out for that now.

Reply:
I've found a way to test this locally with my back-end PR, so this may not be necessary anymore. I'll come back in a few minutes to verify.